xdp.go

```diff
@@ -380,6 +385,50 @@ func NewSocket(Ifindex int, QueueID int, options *SocketOptions) (xsk *Socket, e
 		return nil, fmt.Errorf("syscall.Bind SockaddrXDP failed: %v", err)
 	}
 
+	colle_name := "/home/mydir/bpf.o"
```

Comment:
[xdp.Program](https://pkg.go.dev/github.com/asavie/xdp#Program) is _somewhat_ re-usable, you could perform this `ebpf.LoadCollection` in your own code and then set the `Program`, `Queues`, and `Sockets` maps appropriately -- then the [Program.Attach](https://pkg.go.dev/github.com/asavie/xdp#Program.Attach) and other functions can be re-used.